index.js

```diff
@@ -85,12 +197,20 @@ console.log('A:', a, 'B:', b);
   ...
 ]
 */
+
+console.log("============");
+console.log("Exercise 12:");
+
 const classes = [
-  [ 'Chemistry', '9AM', 'Mr. Darnick' ],
-  [ 'Physics', '10:15AM', 'Mrs. Lithun'],
-  [ 'Math', '11:30AM', 'Mrs. Vitalis' ]
+    ["Chemistry", "9AM", "Mr. Darnick"],
+    ["Physics", "10:15AM", "Mrs. Lithun"],
+    ["Math", "11:30AM", "Mrs. Vitalis"]
 ];
 
-const objClasses = [];
+const key = ["subject", "time", "teacher"];
+
+const objClasses = classes.map(item =>
+    item.reduce((acc, val, ind) => ((acc[key[ind]] = val), acc), {})
```

Comment:
can you maybe make this a little more readable by naming the variables more descriptive?
but very good use of reduce, I like it

Reply:
I did some updates to variable names and inserted some console.log's to highlight the current iteration elements and values, both in .map() and .reduce() methods.